lib/signaling/v2/icemediamonitor.js

```diff
@@ -0,0 +1,98 @@
+/* eslint callback-return:0 */
+'use strict';
+
+const { ICE_MONITOR_ACTIVITY_CHECK_PERIOD_MS, ICE_MONITOR_INACTIVITY_THRESHOLD_MS } = require('../../util/constants');
+
+/**
+ * @emits IceMediaMonitor#updated
+ */
+class IceMediaMonitor {
+  /**
+   * Construct a {@link IceMediaMonitor}.
+   * @param {PeerConnectionV2} pc
+   */
+  constructor(pc, options) {
+    options = Object.assign({
+      activityCheckPeriodMS: ICE_MONITOR_ACTIVITY_CHECK_PERIOD_MS,
+      inactivityThresholdMS: ICE_MONITOR_INACTIVITY_THRESHOLD_MS,
+    }, options);
+
+    Object.defineProperties(this, {
+      _pc: {
+        value: pc
+      },
+      _timer: {
+        value: null,
+        writable: true,
+      },
+      _activityCheckPeriodMS: {
+        value: options.activityCheckPeriodMS
+      },
+      _inactivityThresholdMS: {
+        value: options.inactivityThresholdMS
+      }
+    });
+  }
+
+  // TODO: remove this before merging.
+  _makarandlog() {
+    var newArgs = ['makarand: '];
+    var args = Array.prototype.slice.call(arguments);
+    console.log.apply(console.log, newArgs.concat(args));
+  }
+
+  _getMediaStats() {
+    return this._pc.getMediaStats();
+  }
+  /**
+   * Start monitoring.
+   * @returns {void}
+   */
+  start(onInactiveCallback) {
```

Comment:
I initially had  this class as EventEmitter (instead of taking callback param here), but decided against that, as callback works better in this case because 1) we have just one listener 2) the lifetime of callback is between start and stop, and it's easier for caller to call start/stop instead of an additional on/off for event listener.

Reply:
@makarandp0 I think you can convert this to a Promise-based method and get rid of the `stop()` method 
 and the `_timer` property:
```js
class IceConnectionMonitor {
  ...
  /**
   * Check if the ICE connection is inactive.
   * @returns {Promise<void>}
   */
  checkIfInactive() {
    return wait(this._activityCheckPeriodMs).then(() => {
      return this._getIceConnectionStats();
    }).then(stats => {
      if (!stats) {
        return this.checkIfInactive();
      }
      if (!this._lastActiveStats || this._lastActiveStats.bytesReceived !== stats.bytesReceived) {
        this._lastActiveStats = stats;
      }
      if (stats.timestamp - this._lastActivity.timestamp < this._inactivityThresholdMs) {
        return this.checkIfInactive();
      }
    });
  }
  ...
}

// This can be moved to lib/utils/index.js

/**
 * Wait for a given period of time.
 * @param {number} delayMs
 * @returns {Promise<void>}
 */
function wait(delayMs) {
  return new Promise(resolve => setTimeout(resolve, delayMs));
}
```

Reply:
@manjeshbhargav, I think your suggestion would work if we decided to only start the monitor when Ice goes into disconnected state, and simply wait for 3 seconds of non-activity. But since we are monitoring at connected state, we need this callback to fire every time it detects inactivity, and peerConnection object acts on it only when it finds itself in disconnected state when callback fires. Maybe I can setup a quick meeting to syncup on this.

Reply:
never mind my comment above - I think I better understand your proposal now, and how it would work, but I do have few concerns with it, as I commented earlier.

Reply:
updated this now that we've decided to listen only after disconnect.